### PR TITLE
test: remove Boost Test from libtest_util

### PR DESCRIPTION
**Pull request by fanquake**
Context is the discussion here:
https://github.com/bitcoin/bitcoin/pull/25974/files#r961541457.

Output:
```bash
[test/util/chainstate.h:38] [CreateAndActivateUTXOSnapshot] Wrote UTXO snapshot to /var/folders/sq/z88fhjzj0b19ftsd2_bjrmjm0000gn/T/test_common_Bitcoin Core/8f2783bb3dbf10c669cd892192d70efcca4bab250226856fed7ffecdb378ffc7/test_snapshot.100.dat: {"coins_written":100,"base_hash":"571d80a9967ae599cec0448b0b0ba1cfb606f584d8069bd7166b86854ba7a191","base_height":100,"path":"/var/folders/sq/z88fhjzj0b19ftsd2_bjrmjm0000gn/T/test_common_Bitcoin Core/8f2783bb3dbf10c669cd892192d70efcca4bab250226856fed7ffecdb378ffc7/test_snapshot.100.dat","txoutset_hash":"cd1ba1c3f393058ae743b7c6bdbd00c897744cdcf022c9f2f0f2b4565c08a49c","nchaintx":101}
```

### CODE REVIEW

File: src/test/util/chainstate.h [hunk 1]
```diff
@@ -36,8 +35,8 @@ CreateAndActivateUTXOSnapshot(node::NodeContext& node, const fs::path root, F ma
 
     UniValue result = CreateUTXOSnapshot(
         node, node.chainman->ActiveChainstate(), auto_outfile, snapshot_path, snapshot_path);
-    BOOST_TEST_MESSAGE(
-        "Wrote UTXO snapshot to " << fs::PathToString(snapshot_path.make_preferred()) << ": " << result.write());
+    LogPrintf(
+        "Wrote UTXO snapshot to %s: %s", fs::PathToString(snapshot_path.make_preferred()), result.write());
```
[the remark this answers is not in the document]
Comment (fanquake) [reply src/test/util/chainstate.h:39]: I don't think that's a break in behaviour. The message is still written to the `.log`.

[the remark this answers is not in the document]
Comment (hebasto) [reply src/test/util/chainstate.h:39]: > The message is still written to the `.log`.

That is also a behavior change:

- on master (0ebd4db32b39cb7c505148f090df4b7ac778c307):
```
$ ./src/test/test_bitcoin -t validation_chainstate_tests -- DEBUG_LOG_OUT | grep -e 'Wrote UTXO'

*** No errors detected
```

- this PR (a7dbf74d72abfe74aaeb3c11dbfa98d43b85b4ec):
```
$ ./src/test/test_bitcoin -t validation_chainstate_tests -- DEBUG_LOG_OUT | grep -e 'Wrote UTXO'
2022-09-07T14:26:09.949830Z (mocktime: 2020-08-31T15:34:22Z) [test] [test/util/chainstate.h:39] [CreateAndActivateUTXOSnapshot] Wrote UTXO snapshot to /tmp/test_common_Bitcoin Core/d86bd6189fcd798f749edf44bf5029213c9a8add2bc39da5272b201d821e341d/test_snapshot.110.dat: {"coins_written":110,"base_hash":"696e92821f65549c7ee134edceeeeaaa4105647a3c4fd9f298c0aec0ab50425c","base_height":110,"path":"/tmp/test_common_Bitcoin Core/d86bd6189fcd798f749edf44bf5029213c9a8add2bc39da5272b201d821e341d/test_snapshot.110.dat","txoutset_hash":"1ebbf5850204c0bdb15bf030f47c7fe91d45c44c712697e4509ba67adb01c618","nchaintx":111}[Chainstate [ibd] @ height 110 (696e92821f65549c7ee134edceeeeaaa4105647a3c4fd9f298c0aec0ab50425c)] resized coinsdb cache to 0.0 MiB

*** No errors detected
```

> I don't think that's a break in behaviour.

But behaviour change looks obvious, doesn't it?

**Comment by MarcoFalke** [comment]
For those that are wondering if this function is used in the fuzz tests. Likely no, because it would lead to linker errors if used in the fuzz tests: 

```
/usr/bin/ld: test/fuzz/fuzz-addition_overflow.o: in function `_GLOBAL__sub_I_addition_overflow.cpp':
/usr/include/boost/test/unit_test_log.hpp:227: undefined reference to `boost::unit_test::unit_test_log_t::instance()'
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by theuni** [comment]
ACK a7dbf74d72abfe74aaeb3c11dbfa98d43b85b4ec

Agree this is a slight behavioral change, but as it's the only use of boost it seems like pretty obvious low-hanging fruit.

From #25974:
> Ah ok. I think I have a slight preference to keep boost out of libtest_util, and thus out of the fuzz tests

I think this also means that we could drop boost from the `libtest_util_a_CPPFLAGS` to enforce that new dependencies don't creep back in, no?

Unfortunately, boost is lumped in with `BITCOIN_INCLUDES`, but it's the only external dependency that remains there. Maybe it's time to break `BOOST_CPPFLAGS` out of there and use it where it's needed instead (all over the place I assume, lots of copy/paste), that way we can surgically enforce where it's allowed.

Consider that a suggestion for a follow-up PR :)

**Comment by fanquake** [comment]
> Consider that a suggestion for a follow-up PR :)

Started in #26056.

**Pull request merged**